src/isyntax/isyntax.c

```diff
@@ -3063,20 +3063,22 @@ bool isyntax_open(isyntax_t* isyntax, const char* filename, bool init_allocators
 
 						}
 
-						// When recursively decoding the tiles, at each iteration the image is slightly offset
-						// to the top left.
+                        // When recursively decoding the tiles, at each iteration the image is slightly offset
+                        // to the top left.
                         // The amount of shift seems correspond to the level padding: ((3 >> (scale-1)) - 2).
-						// (I guess this is related to the way the wavelet transform works.)
-						// Put another way: the highest (zoomed out levels) are shifted the to the bottom right
-						// (this is also reflected in the x and y coordinates of the codeblocks in the iSyntax header).
-						for (i32 scale = 1; scale < wsi_image->level_count; ++scale) {
-							isyntax_level_t* level = wsi_image->levels + scale;
+                        // (I guess this is related to the way the wavelet transform works.)
+                        // Put another way: the highest (zoomed out levels) are shifted the to the bottom right
+                        // (this is also reflected in the x and y coordinates of the codeblocks in the iSyntax header).
+                        for (i32 scale = 0; scale < wsi_image->level_count; ++scale) {
+                            isyntax_level_t* level = wsi_image->levels + scale;
+                            level->origin_offset_in_pixels = ((PER_LEVEL_PADDING << wsi_image->level_count) - PER_LEVEL_PADDING) >> scale;
```

Comment:
This doesn't look right (see my other comment elsewhere).
To keep the git history clean I'll fixup and reproduce the changes from this commit in my own commit.